src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,389 @@
+package teammates.storage.api;
+
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.NotificationType;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.test.BaseTestCaseWithLocalDatabaseAccess;
+
+/**
+ * SUT: {@link NotificationsDb}.
+ */
+public class NotificationsDbTest extends BaseTestCaseWithLocalDatabaseAccess {
+
+    // Two sets of attribute values for testing.
+    static final String TITLE = "title";
+    static final String TITLE_DIFF = "title_diff";
+    static final String MESSAGE = "message";
+    static final String MESSAGE_DIFF = "message_diff";
+    static final NotificationType TYPE = NotificationType.DEPRECATION;
+    static final NotificationType TYPE_DIFF = NotificationType.MAINTENANCE;
+    static final NotificationTargetUser TARGET_USER = NotificationTargetUser.STUDENT;
+    static final NotificationTargetUser TARGET_USER_DIFF = NotificationTargetUser.INSTRUCTOR;
+    static final Instant START_TIME = Instant.now();
+    static final Instant START_TIME_DIFF = START_TIME.plusSeconds(600);
+    static final Instant END_TIME = START_TIME.plusSeconds(3600);
+    static final Instant END_TIME_DIFF = END_TIME.plusSeconds(600);
+
+    static final String NON_EXISTENT_ID = "invalid_notification_id";
+
+    private final NotificationsDb notificationsDb = NotificationsDb.inst();
+
+    @Test
+    public void testGetNotification() throws Exception {
+        NotificationAttributes n = createNewNotification();
+
+        ______TS("typical success case");
+        NotificationAttributes retrieved = notificationsDb.getNotification(n.getNotificationId());
+        assertNotNull(retrieved);
+
+        ______TS("expect null for non-existent account");
+        retrieved = notificationsDb.getNotification(NON_EXISTENT_ID);
+        assertNull(retrieved);
+
+        ______TS("failure: null parameter");
+        assertThrows(AssertionError.class, () -> notificationsDb.getNotification(null));
+
+        // delete created account
+        notificationsDb.deleteNotification(n.getNotificationId());
+    }
+
+    @Test
+    public void testGetAllNotifications() throws Exception {
+        ______TS("typical success case");
+        final int numOfNotifications = 10;
+
+        Set<String> notificationIds = new HashSet<>();
+        // Create 10 new notifications and set the occurance to be false.
+        for (int i = 0; i < numOfNotifications; i++) {
+            NotificationAttributes n = createNewNotification();
+            notificationIds.add(n.getNotificationId());
+        }
```

Comment:
We should create setup (`@BeforeMethod`) and teardown (`@AfterMethod`) methods to reduce all the create notification and delete notification boilerplate.

Reply:
Thanks for the comment. I think using `@AfterMethod` to do clean up is a good idea. However, it is probably hard to do initialization using the `@BeforeMethod` as the number/attribute values for each test can be different?

Reply:
I have implemented the `@AfterMethod`. Would you like to take a look at it?

Reply:
For the `@BeforeMethod` you can actually use data from `typicalDataBundle.json`, so that you can vary attribute values for each test notification.

Reply:
For tests which involve static test data e.g. fixed notification `startTime`, `endTime` etc we can use the data from `typicalDataBundle.json`. 

In `testGetAllNotifications`, we only want to test whether it returns all notifications in DB, in this case our setup method will populate the DB with static data and we can just check the size of the list of notifications returned. 

In other tests, static test data might not suffice, e.g. we need to set the `startTime` based on certain offset from now, then we have no choice but to create that notification in the test method itself.

Reply:
Actually all tests use start and end time based on a offset from now... But I guess can use a very old date and a future date as data point. I will take a look at that.

Reply:
I have pushed the latest commit for the change. Data initialization are done through `@BeforeMethod` using static data from `typicalDataBundle` now.